app/src/main/java/net/pp3345/ykdroid/yubikey/UsbYubiKey.java

```diff
@@ -96,6 +109,20 @@ public String getName() {
 		public String getVersion() {
 			return this.version;
 		}
+
+		public static Type lookupDeviceType(final UsbDevice device) {
+			for (final Type type : Type.values()) {
+				if (type.getVendorID() == device.getVendorId() &&
+						type.getProductID() == device.getProductId()) {
+					return type;
+				}
+			}
+			return Type.UNKNOWN;
+		}
+
+		public static boolean isDeviceKnown(final UsbDevice device) {
+			return lookupDeviceType(device) != UNKNOWN;
```

Comment:
Currently, this will match both the vendor ID and the product ID. Can we change this to return true even if only the vendor is known? Otherwise this would require to keep the list of YubiKeys up-to-date (it's outdated already by now...).

Reply:
I've added a new YK_UNKNOWN type, which is only matched on vendor ID.